### ISSUE TITLE
could not find file metadata.json

### ISSUE THREAD

**Issue by PurplePanda395** [ZaneYork/SMAPI-Android-Installer#874]
Okay so first off, I'm using SMAPI 3.18.3.3 which is the latest for Android I think? And stardew valley 1.5.6.39. I'm trying to use the ridgeside village mod (I have all the required mods, etc.) but I think the issue is SMAPI itself, so I don't think the mods matter.

[Here](https://smapi.io/log/034df72c24114798a6f936a06deabe06) is my error log.

I've uninstalled and reinstalled three times, cleared the cache AND storage, force stopped it, and nothing worked. I googled my issue, but all I found was people asking about manifest.json which is close, but not exactly my problem. So... No luck there. Was hoping someone here could help?

**Comment by Gaming-pandatje** [comment]
> Okay so first off, I'm using SMAPI 3.18.3.3 which is the latest for Android I think? And stardew valley 1.5.6.39. I'm trying to use the ridgeside village mod (I have all the required mods, etc.) but I'm pretty sure the issue is SMAPI itself, so I don't think the mods matter.
> 
> [Here](https://smapi.io/log/034df72c24114798a6f936a06deabe06) is my error log.
> 
> I've uninstalled and reinstalled three times, cleared the cache AND storage, force stopped it, and nothing worked. I know absolutely nothing about code, so I googled my issue, but all I found was people asking about manifest.json which is close, but not exactly my problem. So... No luck there. Was hoping someone here could help?

could you tell me where you put your mods?

**Comment by Gaming-pandatje** [comment]
to fix the error I would reccommend to delete the com.zane.stardewvalley folder in the data folder then delete smapi reinstall and open. when oppened you select the old stardew valley folder.

**Comment by PurplePanda395** [comment]
I know, but I can't move them there. I try, but it always comes up with "to protecr your privacy please choose another folder" when I try to get into Android/data. I can view the files that are there with another file manager (I'm using fv file explorer) but I can't move the mods there. They show up in SMAPI's config menu though and the "set mods path" is StardewValley/Mods

**Comment by PurplePanda395** [comment]
> to fix the error I would reccommend to delete the com.zane.stardewvalley folder in the data folder then delete smapi reinstall and open. when oppened you select the old stardew valley folder.

Okay I'll try this!

**Comment by Gaming-pandatje** [comment]
> I know, but I can't move them there. I try, but it always comes up with "to protecr your privacy please choose another folder" when I try to get into Android/data. I can view the files that are there with another file manager (I'm using fv file explorer) but I can't move the mods there. They show up in SMAPI's config menu though and the "set mods path" is StardewValley/Mods

you need acces to the data folder to use mods. I would reccomend using the app called zarchiver.

**Comment by Gaming-pandatje** [comment]
I made a tutoriol on how to android modding you can watch: https://youtu.be/UrbZA92XR-o?si=UPGHppApU5aACrtj

**Comment by PurplePanda395** [comment]
Thank you!

**Comment by PurplePanda395** [comment]
![Screenshot_20241209-100822](https://github.com/user-attachments/assets/6cdb7686-a89b-4aea-975c-79c658c93820)

Do you know how to fix this?

**Comment by Gaming-pandatje** [comment]
What happens when you click on ok?

**Comment by PurplePanda395** [comment]
![Screenshot_20241209-103036](https://github.com/user-attachments/assets/5b729ddf-1d0a-4ad9-bf66-9019408a6c2d)

It just takes me to my normal files app but I still can't access Android/data

**Comment by Gaming-pandatje** [comment]
You will have to install an app called shizuku

**Comment by Gaming-pandatje** [comment]
https://youtu.be/CMVfgsGNEXI

**Comment by Gaming-pandatje** [comment]
did you get it working?

**Comment by PurplePanda395** [comment]
> did you get it working?

Yes but there is no mods folder? Should I just create a new folder and name it "Mods", and put the mods there? Would the path still work?

**Comment by Gaming-pandatje** [comment]
Have you already opened Stardew valley smapi and selected the old Stardew folder in the app?

**Comment by PurplePanda395** [comment]
Yes. Should I uninstall and reinstall SMAPI?

**Comment by Gaming-pandatje** [comment]
> Yes. Should I uninstall and reinstall SMAPI?

Which folders show up in the com.zane.stardewvalley folder?

**Comment by PurplePanda395** [comment]
Files, and then in that it's saves, screenshots, and smapi-internal

**Comment by Gaming-pandatje** [comment]
Have you already opened Stardew valley smapi and selected the old Stardew folder in the app?

Ok, create a folder called mods

**Comment by Gaming-pandatje** [comment]
Create a folder called mods

**Comment by PurplePanda395** [comment]
Ok i did that and I moved the mods into the Mods folder. I'll try running SMAPI now?

**Comment by PurplePanda395** [comment]
> Ok i did that and I moved the mods into the Mods folder. I'll try running SMAPI now?

Uh nevermind I guess lol
![Screenshot_20241209-110629](https://github.com/user-attachments/assets/a91aca58-eac5-4480-8f6d-92705e702744)

**Comment by Gaming-pandatje** [comment]
I have had this problem too. You will have to back up your mods and delete everything. (Don't forget the installer files). This comment explains it.![Screenshot_20241209-191230.png](https://github.com/user-attachments/assets/6d518390-ff2d-4dba-90b6-059970382abe)

**Comment by PurplePanda395** [comment]
Awesome I'll do that now

**Comment by PurplePanda395** [comment]
How do I backup com.zane.stardewvalley? Just save it to my Google drive, or is there something else I do?

**Comment by Gaming-pandatje** [comment]
You don't have to backup the whole folder it just means you have to copy your mods and saves to another folder if you don't have a copy somewhere else.

**Comment by PurplePanda395** [comment]
> You don't have to backup the whole folder it just means you have to copy your mods and saves to another folder if you don't have a copy somewhere else.

Ohhh ok!

**Comment by Gaming-pandatje** [comment]
Hello?

**Comment by Gaming-pandatje** [comment]
Did you use the installer again?

**Comment by PurplePanda395** [comment]
Yes. I followed the instructions on the screenshot you provided

**Comment by Gaming-pandatje** [comment]
Yeah but did you use the installer app to reinstall the scampi files and get the smapi app.

**Comment by PurplePanda395** [comment]
> Yeah but did you use the installer app to reinstall the scampi files and get the smapi app.

Yes

**Comment by Gaming-pandatje** [comment]
Maybe try again because the log says you don't have the smapi internal file and don't forget to delete the Stardew valley file. If this doesn't work I don't know what the problem is and I can't help any further.

**Comment by PurplePanda395** [comment]
> I just copied monogame.framework.dll from StardewValley/smapi-internal and pasted it to com.zane.stardewvalley/files/smapi-internal so I guess I'll see if that works?

I don't think it worked lmao

https://smapi.io/log/80a98e48abbd484eb57d60887589f781

**Comment by PurplePanda395** [comment]
Okay so I uninstalled and reinstalled SMAPI and it works, kinda. There's no errors really, but it freezes right before the title screen
![Screenshot_20241209-125855](https://github.com/user-attachments/assets/539a099f-2726-4547-a162-2d742f29083b)

**Comment by PurplePanda395** [comment]
Huh. There's no error but it still freezes. It doesn't crash, it just stays frozen until I get tired of looking at it and close the game lol

https://smapi.io/log/29d7828b2f7b43c19d2ce4e8f237833d

**Comment by PurplePanda395** [comment]
Ooh okay so I figured out that the music.xwb file is corrupted and that's why it's not working and since I've tried uninstalling and reinstalling the 1.5.6.39 stardew APK multiple times, and it still doesn't work, I'm going to try and find a different .39 APK and see if that works

**Comment by PurplePanda395** [comment]
Hmm I couldn't find many .39 apks (there were tons that SAID they were .39, but ended up actually being .52) but the ones I did find and download, the music.xwb was still corrupted so maybe my ohone is doing something to it?? It seems like too much of a coincidence for the same file in every APK I downloaded to be corrupted

**Comment by Grimface437** [comment]
Hey, i can help you, when u try to reinstall, some mods is still there (hidden files) u need to use ur original file explorer to expose hidden file and delete them

**Comment by Grimface437** [comment]
I'm sorry for bad English
